Vlv2TbltDevicePkg/CustomShellPkg/Application/Shell/Shell.c

```diff
@@ -314,6 +316,220 @@ InternalEfiShellStartCtrlSMonitor(
   return (Status);
 }
 
+/**
+  Internal worker function to load and run an image via memory-mapped device path.
+  It is slightly modified InternalShellExecuteDevicePath() from ShellProtocol.c.
+
+  @param ParentImageHandle      A handle of the image that is executing the specified
+                                command line.
+  @param DevicePath             device path of the file to execute
+  @param CommandLine            Points to the NULL-terminated UCS-2 encoded string
+                                containing the command line. If NULL then the command-
+                                line will be empty.
+  @param Environment            Points to a NULL-terminated array of environment
+                                variables with the format 'x=y', where x is the
+                                environment variable name and y is the value. If this
+                                is NULL, then the current shell environment is used.
+
+  @param[out] StartImageStatus  Returned status from gBS->StartImage.
+
+  @retval EFI_SUCCESS           The command executed successfully. The status code
+                                returned by the command is pointed to by StatusCode.
+  @retval EFI_INVALID_PARAMETER The parameters are invalid.
+  @retval EFI_OUT_OF_RESOURCES  Out of resources.
+  @retval EFI_UNSUPPORTED       Nested shell invocations are not allowed.
+**/
+EFI_STATUS
+ShellExecuteMemoryDevicePath(
+  IN CONST EFI_HANDLE               *ParentImageHandle,
+  IN CONST MEMMAP_DEVICE_PATH       *DevicePath,
+  IN CONST CHAR16                   *CommandLine OPTIONAL,
+  IN CONST CHAR16                   **Environment OPTIONAL,
+  OUT EFI_STATUS                    *StartImageStatus OPTIONAL
+  )
+{
+  EFI_STATUS                    Status;
+  EFI_STATUS                    StartStatus;
+  EFI_STATUS                    CleanupStatus;
+  EFI_HANDLE                    NewHandle;
+  EFI_LOADED_IMAGE_PROTOCOL     *LoadedImage;
+  LIST_ENTRY                    OrigEnvs;
+  EFI_SHELL_PARAMETERS_PROTOCOL ShellParamsProtocol;
+  CHAR16                        *ImagePath;
+  UINTN                         Index;
+  CHAR16                        *Walker;
+  CHAR16                        *NewCmdLine;
+
+  if (ParentImageHandle == NULL) {
+    return (EFI_INVALID_PARAMETER);
```

Comment:
Why we are not consistent?

```suggestion
    return EFI_INVALID_PARAMETER;
```